### ISSUE TITLE
[Bug]: [benchmark][cluster] Milvus hang when loading 1billion vector

### ISSUE THREAD

**Issue by wangting0128** [milvus-io/milvus#12301]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20211125-2ee0c7c
- Deployment mode(standalone or cluster):cluster
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus-2.0.0rc9.dev7
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

client pod：benchmark-tag-zn4gs-470224986

client log：
```
[2021-11-26 00:47:44,966] [    INFO] - {'total_time': 38271.35, 'rps': 26129.21, 'ni_time': 1.91} (milvus_benchmark.runners.base:151)
[2021-11-26 00:47:45,051] [   DEBUG] - Start flush. (milvus_benchmark.runners.locust:428)
[2021-11-26 00:47:51,416] [   DEBUG] - Milvus flush run in 6.36s (milvus_benchmark.client:52)
[2021-11-26 00:47:51,417] [   DEBUG] - Fulsh done, during time: 6.37 (milvus_benchmark.runners.locust:431)
[2021-11-26 00:47:51,421] [   DEBUG] - Row count: 1000000000 in collection: <sift_1b_128_l2> (milvus_benchmark.client:416)
[2021-11-26 00:47:51,421] [   DEBUG] - 1000000000 (milvus_benchmark.runners.locust:432)
[2021-11-26 00:47:51,422] [   DEBUG] - Start build index for last file (milvus_benchmark.runners.locust:434)
[2021-11-26 00:47:51,422] [    INFO] - Building index start, collection_name: sift_1b_128_l2, index_type: IVF_SQ8, metric_type: L2 (milvus_benchmark.client:273)
[2021-11-26 00:47:51,423] [    INFO] - {'nlist': 1024} (milvus_benchmark.client:275)
[2021-11-26 00:47:51,423] [   DEBUG] - collection: sift_1b_128_l2 Index params: {'index_type': 'IVF_SQ8', 'metric_type': 'L2', 'params': {'nlist': 1024}} (milvus_benchmark.client:28
1)
[2021-11-26 00:48:15,676] [   DEBUG] - Building index done, collection_name: sift_1b_128_l2, response: Status(code=0, message='') (milvus_benchmark.client:283)
[2021-11-26 00:48:15,677] [   DEBUG] - Milvus create_index run in 24.25s (milvus_benchmark.client:52)
[2021-11-26 00:48:15,677] [   DEBUG] - {'flush_time': 6.37, 'build_time': 24.26} (milvus_benchmark.runners.locust:438)
[2021-11-26 00:48:15,681] [   DEBUG] - Row count: 1000000000 in collection: <sift_1b_128_l2> (milvus_benchmark.client:416)
[2021-11-26 00:48:15,681] [    INFO] - 1000000000 (milvus_benchmark.runners.locust:439)
[2021-11-26 00:48:15,682] [    INFO] - Start load collection (milvus_benchmark.runners.locust:440)
```

### Expected Behavior

argo task：benchmark-tag-zn4gs

test yaml：
client-configmap: client-random-locust-search-84h-1b
server-configmap: server-cluster-8c64m-datanode2-indexnode4-querynode6

server：
```
NAME                                                         READY   STATUS      RESTARTS   AGE     IP             NODE                      NOMINATED NODE   READINESS GATES
benchmark-tag-zn4gs-1-etcd-0                                 1/1     Running     0          23h     10.97.17.253   qa-node014.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-etcd-1                                 1/1     Running     0          23h     10.97.17.3     qa-node014.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-etcd-2                                 1/1     Running     0          23h     10.97.17.254   qa-node014.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-datacoord-56d8bc7c8d-wsztr      1/1     Running     0          23h     10.97.8.12     qa-node006.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-datanode-5c8bd8b4fc-2tw5n       1/1     Running     0          23h     10.97.17.247   qa-node014.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-datanode-5c8bd8b4fc-cqqw5       1/1     Running     0          23h     10.97.14.44    qa-node011.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-indexcoord-6f87b4d64f-6n8zv     1/1     Running     0          23h     10.97.7.154    qa-node005.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-indexnode-54778dd9b-4cdg8       1/1     Running     0          23h     10.97.10.125   qa-node008.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-indexnode-54778dd9b-94jqn       1/1     Running     0          23h     10.97.17.249   qa-node014.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-indexnode-54778dd9b-c66qd       1/1     Running     0          23h     10.97.15.53    qa-node012.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-indexnode-54778dd9b-t6cpr       1/1     Running     0          23h     10.97.17.248   qa-node014.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-proxy-5fb84d4686-7t6mz          1/1     Running     0          23h     10.97.8.11     qa-node006.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-querycoord-678db58c4-hbrvb      1/1     Running     0          23h     10.97.7.152    qa-node005.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-querynode-56cfc64458-5xf46      1/1     Running     0          23h     10.97.19.159   qa-node016.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-querynode-56cfc64458-fcnht      1/1     Running     0          23h     10.97.10.126   qa-node008.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-querynode-56cfc64458-n2htq      1/1     Running     0          23h     10.97.14.45    qa-node011.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-querynode-56cfc64458-rggnl      1/1     Running     0          23h     10.97.11.156   qa-node009.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-querynode-56cfc64458-tb8gh      1/1     Running     0          23h     10.97.13.55    qa-node010.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-querynode-56cfc64458-w2grn      1/1     Running     0          23h     10.97.15.54    qa-node012.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-milvus-rootcoord-6d9f96f68d-7gxfw      1/1     Running     0          23h     10.97.9.120    qa-node007.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-minio-0                                1/1     Running     0          23h     10.97.7.157    qa-node005.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-minio-1                                1/1     Running     0          23h     10.97.7.156    qa-node005.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-minio-2                                1/1     Running     0          23h     10.97.4.93     qa-node002.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-minio-3                                1/1     Running     0          23h     10.97.4.94     qa-node002.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-pulsar-autorecovery-7976b7c784-kxt7r   1/1     Running     0          23h     10.97.14.43    qa-node011.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-pulsar-bastion-7f994456b4-mbchl        1/1     Running     0          23h     10.97.9.119    qa-node007.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-pulsar-bookkeeper-0                    1/1     Running     3          23h     10.97.8.13     qa-node006.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-pulsar-bookkeeper-1                    1/1     Running     2          23h     10.97.4.95     qa-node002.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-pulsar-broker-7bd5c8f97f-6m79c         1/1     Running     0          23h     10.97.9.118    qa-node007.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-pulsar-proxy-558bc45947-hdzh9          2/2     Running     0          23h     10.97.15.52    qa-node012.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-pulsar-zookeeper-0                     1/1     Running     0          23h     10.97.7.155    qa-node005.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-pulsar-zookeeper-1                     1/1     Running     0          23h     10.97.7.158    qa-node005.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-pulsar-zookeeper-2                     1/1     Running     0          23h     10.97.7.161    qa-node005.zilliz.local   <none>           <none>
benchmark-tag-zn4gs-1-pulsar-zookeeper-metadata-mtmx7        0/1     Completed   0          23h     10.97.7.153    qa-node005.zilliz.local   <none>           <none>
```

**Comment by xige-16** [comment]
a watchDeltaChannel task hang on querynode
[benchmark-tag-zn4gs-1-milvus-querynode-56cfc64458-n2htq.log](https://github.com/milvus-io/milvus/files/7607255/benchmark-tag-zn4gs-1-milvus-querynode-56cfc64458-n2htq.log)

**Comment by godchen0212** [comment]
We discover the pulsar bookeeper crashed for a minute at time 02:10-02:11.
And the reader reads data from the checkpoint at time 02:10.
It makes ``msgstream.Next`` blocking, because it is a blocking interface, it will read the data until data is read. We expect that after pulsar resumes , the ``Next`` function will return, but it does not do so.
We will add a timeout for next function, if he does not read the data within 10 seconds, it returns failure.

**Comment by godchen0212** [comment]
This problem has been fixed in pr #12308. Please check it again.
/unassign
/assign @wangting0128

**Comment by xige-16** [comment]
reproduce in #12456
